app/jobs/google/search_keyword_job.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Google
+  class ClientServiceError < StandardError; end
+
+  class SearchKeywordJob < ApplicationJob
+    queue_as :default
+
+    retry_on ClientServiceError, ArgumentError
+
+    def perform(keyword_id)
+      keyword = Keyword.where(id: keyword_id).select(:id, :name, :user_id).take
+
+      return unless keyword
+
+      html_result = ClientService.new(keyword: keyword.name).call
+
+      raise ClientServiceError unless html_result
+
+      update_keyword keyword_id, ParserService.new(html_response: html_result).call
+    rescue ActiveRecord::RecordNotFound, ClientServiceError, ArgumentError
```

Comment:
Thi this we don't need this `rescue block. If any exception raises, then sidekiq will `retry` again and `keyword` status will be the same it is of the exception raise, So I this k we don't need to update to again to  `failed`. What do you think?

Reply:
I'm not sure to understand... 
The only purpose of this rescue is to update the status from :pending to :fail

Do you mean to just keep keywords "pending" even when they failed? It is possible, that would simplify the code, but my initial idea was to notify the user when a keyword failed.

If we do not have a `:failed` status, then maybe we should simply remove the whole column, no? Simply check if a `count` column is NIL will tell us if pending or parsed. 

Let me know what you think? 
Personally I would prefer keeping the 'failed' status~ But this is just my opinion, I'm open to change ~

Reply:
I got it, you wanted to keep them in `failed` state. Then its fine to me